{"kind":"symbol","primaryContentSections":[{"declarations":[{"tokens":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"text":"awaitWithoutException","kind":"identifier"},{"text":"(","kind":"text"},{"text":"millis","kind":"externalParam"},{"text":" ","kind":"text"},{"kind":"internalParam","text":"timeOut"},{"kind":"text","text":": "},{"text":"Int32","kind":"typeIdentifier","preciseIdentifier":"s:s5Int32V"},{"kind":"text","text":") -> "},{"text":"Bool","kind":"typeIdentifier","preciseIdentifier":"s:Sb"}],"platforms":["macOS"],"languages":["swift"]}],"kind":"declarations"},{"content":[{"text":"Return Value","anchor":"return-value","level":2,"type":"heading"},{"type":"paragraph","inlineContent":[{"text":"If the wait times out, then the computation is ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","isActive":true,"type":"reference"},{"type":"text","text":" cancelled and this method returns "},{"type":"strong","inlineContent":[{"text":"false","type":"text"}]},{"text":".","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"Use this method in the code that shall continue normal execution in case of timeout."}]}],"kind":"content"}],"metadata":{"symbolKind":"method","title":"awaitWithoutException(millis:)","modules":[{"name":"DXFeedFramework"}],"externalID":"s:15DXFeedFramework7PromiseC21awaitWithoutException6millisSbs5Int32V_tF","role":"symbol","fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"awaitWithoutException"},{"kind":"text","text":"("},{"kind":"externalParam","text":"millis"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5Int32V","kind":"typeIdentifier","text":"Int32"},{"text":") -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Sb","text":"Bool"}],"roleHeading":"Instance Method"},"abstract":[{"text":"Wait for computation to complete and return its result or throw an exception in case of exceptional completion.","type":"text"},{"text":" ","type":"text"},{"text":"If the wait times out, then the computation is ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","isActive":true,"type":"reference"},{"text":" cancelled and exception is thrown.","type":"text"}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/awaitWithoutException(millis:)"},"schemaVersion":{"minor":3,"major":0,"patch":0},"hierarchy":{"paths":[["doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise"]]},"sections":[],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/dxfeedframework\/promise\/awaitwithoutexception(millis:)"]}],"references":{"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isDone()":{"type":"topic","abstract":[{"type":"text","text":"Returns "},{"type":"strong","inlineContent":[{"type":"text","text":"true"}]},{"type":"text","text":" when computation has"},{"text":" ","type":"text"},{"type":"reference","isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)"},{"type":"text","text":" completed normally,"},{"text":" ","type":"text"},{"type":"text","text":"or "},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","type":"reference","isActive":true},{"type":"text","text":" exceptionally,"},{"type":"text","text":" "},{"text":"or was ","type":"text"},{"type":"reference","isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()"},{"text":" canceled","type":"text"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","title":"isDone()","role":"symbol","kind":"symbol","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"isDone","kind":"identifier"},{"text":"() -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Sb","text":"Bool"}],"url":"\/documentation\/dxfeedframework\/promise\/isdone()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/awaitWithoutException(millis:)":{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/awaitWithoutException(millis:)","url":"\/documentation\/dxfeedframework\/promise\/awaitwithoutexception(millis:)","title":"awaitWithoutException(millis:)","type":"topic","fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"awaitWithoutException"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"millis"},{"kind":"text","text":": "},{"text":"Int32","kind":"typeIdentifier","preciseIdentifier":"s:s5Int32V"},{"text":") -> ","kind":"text"},{"preciseIdentifier":"s:Sb","text":"Bool","kind":"typeIdentifier"}],"kind":"symbol","role":"symbol","abstract":[{"type":"text","text":"Wait for computation to complete and return its result or throw an exception in case of exceptional completion."},{"text":" ","type":"text"},{"type":"text","text":"If the wait times out, then the computation is "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()"},{"text":" cancelled and exception is thrown.","type":"text"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise":{"fragments":[{"text":"class","kind":"keyword"},{"kind":"text","text":" "},{"text":"Promise","kind":"identifier"}],"title":"Promise","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise","role":"symbol","url":"\/documentation\/dxfeedframework\/promise","type":"topic","abstract":[{"text":"Result of a computation that will be completed normally or exceptionally in the future.","type":"text"},{"type":"text","text":" "},{"type":"text","text":"This class is designed to represent a promise to deliver certain result."}],"kind":"symbol","navigatorTitle":[{"text":"Promise","kind":"identifier"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/cancel()":{"type":"topic","abstract":[{"type":"text","text":"Cancels computation. This method does nothing if computation has already "},{"type":"reference","isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()"},{"text":" completed.","type":"text"}],"title":"cancel()","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","role":"symbol","kind":"symbol","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"kind":"identifier","text":"cancel"},{"text":"()","kind":"text"}],"url":"\/documentation\/dxfeedframework\/promise\/cancel()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework":{"type":"topic","abstract":[],"kind":"symbol","url":"\/documentation\/dxfeedframework","role":"collection","title":"DXFeedFramework","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework"}}}